cloud/db_cloud_test.cc

```diff
@@ -2654,30 +2648,35 @@ TEST_F(CloudTest, CloudManifestFileDeletionTest) {
   OpenDB();
   CloseDB();
 
+  auto checkCloudManifestFileExistence = [&](std::vector<std::string> cookies) {
+    for (auto cookie : cookies) {
+      EXPECT_OK(
+          GetCloudFileSystemImpl()->GetStorageProvider()->ExistsCloudObject(
+              GetCloudFileSystemImpl()->GetDestBucketName(),
+              MakeCloudManifestFile(
+                  GetCloudFileSystemImpl()->GetDestObjectPath(), cookie)));
+    }
+  };
+
   // double check that the CM files are indeed created
-  for (auto cookie: {"", "1"}) {
-    EXPECT_OK(GetCloudFileSystemImpl()->GetStorageProvider()->ExistsCloudObject(
-        GetCloudFileSystemImpl()->GetDestBucketName(),
-        MakeCloudManifestFile(GetCloudFileSystemImpl()->GetDestObjectPath(),
-                              cookie)));
-  }
+  checkCloudManifestFileExistence({"", "1"});
 
   // set large file deletion delay so that files are not deleted immediately
-  GetCloudFileSystemImpl()->TEST_SetFileDeletionDelay(std::chrono::hours(1));
+  cloud_fs_options_.cloud_file_deletion_delay = std::chrono::hours(1);
+  EXPECT_EQ(GetCloudFileSystemImpl()->TEST_NumScheduledJobs(), 0);
 
   // now we reopen the db with empty cookie_on_open and new_cookie_on_open =
   // "1". Double check that CLOUDMANIFEST-1 is not deleted!
   OpenDB();
-  EXPECT_EQ(GetCloudFileSystemImpl()->TEST_NumScheduledJobs(), 0);
+  checkCloudManifestFileExistence({"", "1"});
```

Comment:
This test is updated since it actually didn't work as expected. `TEST_SetFileDeletionDelay` at line 2666 doesn't work since the `CloudFileSystem` is always recreated when calling `OpenDB`.